### ISSUE TITLE
Typescript typing is broken in version 1.0.0

### ISSUE THREAD

**Issue by amakhrov** [GetStream/stream-chat-js#86]
Following errors are generated when trying to compile a Typescript project after importing stream-chat:
```
ERROR in node_modules/stream-chat/types/stream-chat/index.d.ts(45,18): error TS2430: Interface 'MessageResponse' incorrectly extends interface 'Message'.
  Types of property 'mentioned_users' are incompatible.
    Type 'string[]' is not assignable to type 'User[]'.
      Type 'string' is not assignable to type 'User'.
node_modules/stream-chat/types/stream-chat/index.d.ts(98,8): error TS2304: Cannot find name 'OwnUser'.
node_modules/stream-chat/types/stream-chat/index.d.ts(105,3): error TS2300: Duplicate identifier 'type'.
node_modules/stream-chat/types/stream-chat/index.d.ts(109,3): error TS2300: Duplicate identifier 'type'.
node_modules/stream-chat/types/stream-chat/index.d.ts(303,41): error TS2304: Cannot find name 'Immutable'.
node_modules/stream-chat/types/stream-chat/index.d.ts(315,29): error TS2314: Generic type 'Promise<T>' requires 1 type argument(s).
```

I'm using Typescript 3.2

**Comment by vishalnarkhede** [comment]
Hey @amakhrov 

It is now fixed in v1.0.1. Thanks for raising this :)

**Comment by amakhrov** [comment]
That was fast, thanks! :)
